src/appmake/ti8xk.c

```diff
@@ -0,0 +1,547 @@
+/** It should be notes that is is for compling FLASH APPS not ram programs.
+ *  For ram programs see tixx.c
+ * 
+ *  This file originally made by HeronErin. (github.com/HeronErin)
+ * 
+ * 
+ * 
+ * *Heavily* Based on https://github.com/alberthdev/spasm-ng/blob/master/export.cpp
+ * All credit to spasm for most of this 
+ *
+ *
+ */ 
+
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
+#if defined(__APPLE__)
+#  define COMMON_DIGEST_FOR_OPENSSL
+#  include <CommonCrypto/CommonDigest.h>
+#  define SHA1 CC_SHA1
```

Comment:
MD5 is used in this file, so this should be:

```
#define MD5 CC_MD5
```

Reply:
I'm impressed that you got this to work! I'm not keen on the number of custom annotations and sccz80 changes you had to make so it would be nice to reduce them and allow use of the other compilers - I just need to figure out how.

I understand the need to compile banks to bin files - it looks like the paging isn't entirely under user control so we can't do our normal trick of a common/home and a banked page (which does make things simpler and allows more flexible interbank calling).

I'm putting down my thoughts below, it won't work at the moment - I do need to figure out how to get the BCALL target exports working nicely.

## crt0 extension

Introduce a link time variable eg `CRT0_BCALL_TABLE_SIZE` which defines the amount of reserved space for bcalls. Default it to 10 or something like that.

Add the following for each page:

```
    SECTION BANK_1
    org 0x014000
    SECTION CODE_1
    SECTION RODATA_1
    SECTION DATA_1
    SECTION BSS_1
    SECTION BANK_1_END
```

## Compiling files

Use `#pragma bank NN` in the file to move the address to 0x00014000, 0x00024000 as appropriate. This matches up with the way banking works on other machines.

Generate the binary with the -g flag which generates a defc file so we know the address of functions (**problem**: this contains all symbols, not just the ones we want to export).

The theory is that we should be able to include that .defc file in later compilations so we get the address of functions.  **Note** this won't work at the moment. 

## Function decorators

* If we use the decorator `__banked` then function calls are generated with:
```
call banked_call
defw addr
defw bank
```

We can add a flag which switches this to a rst:

```
rst $28
defw addr
defw bank
```

I'll need to come up with a way to add a function epilogue/prologue call to sort out the stack.


## Appmake extensions

- Search the binary files for the rst opcode. Pick up the address that follows and check that it exists in any of the `.map` files. Insert into table. Check table hasn't overflowed. Fix up call site (note, this wastes 2 bytes).